symbology/create-symbols-from-web-styles/src/main/java/com/esri/samples/create_symbols_from_web_styles/CreateSymbolsFromWebStylesSample.java

```diff
@@ -0,0 +1,316 @@
+      symbolStyle.addDoneLoadingListener(() -> {
+        if (symbolStyle.getLoadStatus() == LoadStatus.FAILED_TO_LOAD) {
+          new Alert(Alert.AlertType.ERROR, "Error: could not load symbol style. Details: \n"
+            + symbolStyle.getLoadError().getMessage()).show();
+        }
+      });
+
+      // setup the UI for the legend
+      setupLegend();
+
+      // create a list of the required symbol names from the web style
+      ArrayList<String> symbolNames = new ArrayList<>();
+      symbolNames.addAll(Arrays.asList("atm", "beach", "campground", "city-hall", "hospital", "library", "park",
+        "place-of-worship", "police-station", "post-office", "school", "trail"));
+
+      for (String symbolName : symbolNames) {
+
+        // search for each symbol in the symbol style
+        List<String> searchKey = new ArrayList<>();
+        searchKey.add(symbolName);
+        ListenableFuture<Symbol> searchResult = symbolStyle.getSymbolAsync(searchKey);
+        searchResult.addDoneListener(() -> {
+          try {
+            // get the symbol from the search result
+            Symbol symbol = searchResult.get();
+
+            // get a list of all categories to be mapped to the symbol
+            List<String> categories = mapSymbolNameToField(symbolName);
+
+            for (String category : categories) {
+
+              // create a unique value for each category
+              List<Object> categoryValue = new ArrayList<>();
+              categoryValue.add(category);
+              UniqueValueRenderer.UniqueValue uniqueValue = new UniqueValueRenderer.UniqueValue(
+                "", symbolName, symbol, categoryValue);
+
+              // add each unique value to the unique value renderer
+              uniqueValueRenderer.getUniqueValues().add(uniqueValue);
+            }
+
+            // create and add an image view and a label for the symbol to the legend on the UI
+            ImageView imageView = createImageView(symbol);
+            Label gridPaneLabel = new Label(symbolName);
+            gridPane.add(imageView, 0, symbolNames.indexOf(symbolName) + 2);
+            gridPane.add(gridPaneLabel, 1, symbolNames.indexOf(symbolName) + 2);
+
+          } catch (Exception e) {
+            // on any error, display the stack trace.
+            e.printStackTrace();
+          }
+        });
+      }
+
+      // add a map scale changed listener on the map view to control the symbol sizes at different scales
+      mapView.addMapScaleChangedListener(mapScaleChangedEvent -> {
+        featureLayer.setScaleSymbols(!(mapView.getMapScale() < 80000));
+      });
+
+      // add the map view and grid pane to the stack pane
+      stackPane.getChildren().addAll(mapView, gridPane);
+      StackPane.setAlignment(gridPane, Pos.TOP_LEFT);
+      StackPane.setMargin(gridPane, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
```

Comment:
since this is at the end of the larger try/catch block for the `start` method, I'm happy with leaving this as generic and for the exception to be printed within the console log for the developer to debug. 

Where we are being more explicit is for showing dialog pop up messages, where an end user who isn't a developer may wonder why something didn't load (e.g. feature layer not loading due to no internet connection).